src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9475,54 +9533,59 @@ namespace SurfaceGeometry {
-                        }
-                        ++state.dataErrTracking->TotalCoincidentVertices;
-                        if (nSides > 3) {
-                            if (state.dataGlobal->DisplayExtraWarnings) {
-                                ShowContinueError(state, format("Dropping Vertex [{}].", nextVertexIndex));
-                            }
-                            --nSides;
-                            vertices.erase(itnext);
-                            poppedVertex = true;
-                            break;
-                        } else {
-                            if (state.dataGlobal->DisplayExtraWarnings) {
-                                ShowContinueError(state,
-                                                  format("Cannot Drop Vertex [{}]; Number of Surface Sides at minimum. This surface is now a "
-                                                         "degenerate surface.",
-                                                         curVertexIndex));
-                            }
-                            ++state.dataErrTracking->TotalDegenerateSurfaces;
-                            // mark degenerate surface?
-                        }
+                auto itKept = vertices.begin();
+                std::advance(itKept, popResult.keptVertexPos);
+                int const keptVertexIndex = popResult.keptVertexPos + 1;
+
+                if (state.dataGlobal->DisplayExtraWarnings) {
+                    ShowWarningError(state,
+                                     format("{}Distance between two vertices < .01, possibly coincident. for Surface={}, in Zone={}",
+                                            RoutineName,
+                                            state.dataSurfaceGeometry->SurfaceTmp(SurfNum).Name,
+                                            state.dataSurfaceGeometry->SurfaceTmp(SurfNum).ZoneName));
+
+                    bool const printPoppedFirst = (poppedVertexIndex < keptVertexIndex) ? !(poppedVertexIndex == 1 and keptVertexIndex == nSides)
+                                                                                        : (poppedVertexIndex == nSides and keptVertexIndex == 1);
+
+                    if (printPoppedFirst) {
+                        ShowContinueError(state, format("Vertex [{}]=({:.2R},{:.2R},{:.2R})", poppedVertexIndex, it->x, it->y, it->z));
+                        ShowContinueError(state, format("Vertex [{}]=({:.2R},{:.2R},{:.2R})", keptVertexIndex, itKept->x, itKept->y, itKept->z));
                     } else {
-                        Perimeter += DistanceCheck;
+                        ShowContinueError(state, format("Vertex [{}]=({:.2R},{:.2R},{:.2R})", keptVertexIndex, itKept->x, itKept->y, itKept->z));
+                        ShowContinueError(state, format("Vertex [{}]=({:.2R},{:.2R},{:.2R})", poppedVertexIndex, it->x, it->y, it->z));
+                    }
```

Comment:
Gotta figure out which one of the two to print first. I aim to keep a consistent, natural, order, even when it "wraps around".

Given a surface with 3 vertices, this is what I expect

| poppedVertexIndex | keptVertexIndex | expectedPrintPoppedFirst |
|-------------------|-----------------|--------------------------|
| 1                 | 2               | TRUE                     |
| 2                 | 3               | TRUE                     |
| 3                 | 1               | TRUE                     |
| 1                 | 3               | FALSE                    |
| 2                 | 1               | FALSE                    |
| 3                 | 2               | FALSE                    |